File: src-tauri/src/models/restore_result.rs
use super::{StoredEvent, StoredSnapshot};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RestoreResult {
    pub branch_id: String,
    pub target_version: i64,
    pub snapshot: Option<StoredSnapshot>,
    pub events: Vec<StoredEvent>,
}

File: src/domain/schema/restore-result.ts
import { Schema } from 'effect'
import { StoredEvent } from '~/domain/schema/stored-event'
import { StoredSnapshot } from '~/domain/schema/stored-snapshot'

export const RestoreResult = Schema.Struct({
  branchId: Schema.String,
  targetVersion: Schema.Number,
  snapshot: Schema.NullOr(StoredSnapshot),
  events: Schema.Array(StoredEvent),
})
export type RestoreResult = typeof RestoreResult.Type
